src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonDictionaryDefaultConverter.cs

```diff
@@ -0,0 +1,311 @@
+                            {
+                                JsonSerializer.ThrowUnexpectedMetadataException(propertyName, ref reader, ref state);
+                            }
+                        }
+
+                        state.Current.JsonPropertyNameAsString = reader.GetString();
+                    }
+
+                    if (state.Current.PropertyState < StackFramePropertyState.ReadValue)
+                    {
+                        state.Current.PropertyState = StackFramePropertyState.ReadValue;
+
+                        if (!SingleValueReadWithReadAhead(elementConverter.ClassType, ref reader, ref state))
+                        {
+                            value = default!;
+                            return false;
+                        }
+                    }
+
+                    if (state.Current.PropertyState < StackFramePropertyState.TryRead)
+                    {
+                        // Read the value and add.
+                        bool success = elementConverter.TryRead(ref reader, typeof(TValue), options, ref state, out TValue element);
+                        if (!success)
+                        {
+                            value = default!;
+                            return false;
+                        }
+
+                        Add(element, options, ref state);
+                        state.Current.EndElement();
+                    }
+                }
+            }
+
+            ConvertCollection(ref state, options);
+            value = (TCollection)state.Current.ReturnValue!;
+            return true;
+        }
+
+        internal override sealed bool OnTryWrite(
+            Utf8JsonWriter writer,
+            TCollection dictionary,
+            JsonSerializerOptions options,
+            ref WriteStack state)
+        {
+            bool success;
+
+            if (dictionary == null)
+            {
+                writer.WriteNullValue();
+                success = true;
+            }
+            else
+            {
+                if (!state.Current.ProcessedStartToken)
+                {
+                    state.Current.ProcessedStartToken = true;
+                    writer.WriteStartObject();
+
+                    if (options.ReferenceHandling.ShouldWritePreservedReferences())
+                    {
+                        if (JsonSerializer.WriteReferenceForObject(this, dictionary, ref state, writer) == MetadataPropertyName.Ref)
+                        {
+                            return true;
+                        }
+                    }
+
+                    state.Current.DeclaredJsonPropertyInfo = state.Current.JsonClassInfo.ElementClassInfo!.PolicyProperty!;
+                }
+
+                success = OnWriteResume(writer, dictionary, options, ref state);
+                if (success)
+                {
+                    if (!state.Current.ProcessedEndToken)
```

Comment:
What is this used for? 
I ran the tests removing the usage of this flag on all the converters and nothing broke. 

Maybe we are lacking tests for continuation?

Reply:
Also I don't quite understand the continuation concept while writing. 
I don't see checks that makes you return once you exceed something... 
On the hot path call stack I am only seeing TryWrite -> OnTryWrite -> OnWriteResome -> TryWrite (again). and none of those returns false for dictionary.

Maybe you are missing `if (ShouldFlush) return false;` for  this converter?

Reply:
Yes we are missing tests for continuation against all of the collection types. I stated that in the PR description and @layomia stated he will address. I will create an issue once this PR is in.

I will add the `ShouldFlush` check in case there is a large dictionary of primitives which will make the semantics consistent with the non-dictionary collections. As stated, we need tests to verify that flushing is happening before the buffer is full and expands.